stardog/connection.py

```diff
@@ -778,6 +787,50 @@ def convert(self, content, graph_uri=None):
 
             return r.text
 
+    def report(self, **kwargs):
+        """
+        Produces a SHACL validation report.
+
+        Args (dict):
+          shapes (str, optional): SHACL shapes to validate
+          shacl.shape.graphs (str, optional): SHACL shape graphs to validate
+          nodes (str, optional): SHACL focus node(s) to validate
+          countLimit (str, optional): Maximum number of violations to report
+          shacl.targetClass.simple (boolean, optional): If true, sh:targetClass will be evaluated based on rdf:type triples only, without following rdfs:subClassOf relations
+          shacl.violation.limit.shape (str, optional): number of violation limits per SHACL shapes
+          graph-uri (str, optional): Named Graph
+          reasoning (boolean, optional): Enable Reasoning
+
+
+        Returns:
+          str: SHACL validation report
+
+        Examples:
+          >>> icv.report()
+        """
+
+        accepted_args = [
+            "shapes",
+            "shacl.shape.graphs",
+            "nodes",
+            "countLimit",
+            "shacl.targetClass.simple",
+            "shacl.violation.limit.shape",
+            "graph-uri",
+            "reasoning",
+        ]
+        for arg in kwargs:
+            if arg not in accepted_args:
+                raise Exception("Parameter not recognized")
+
+        url = "/icv/report"
+        kwargs["prettify"] = True
+        params = urllib.parse.urlencode(kwargs)
+        url = f"{url}?{params}"
```

Comment:
Nit: instead of redefining url here, I'd remove the first variable and just set it here `url = f"/icv/report?{params}"` it is always strange to have a variable reference itself. (it really should be illegal but 🤷🏼 )